### ISSUE TITLE
startmenu shortcuts specified as list should default to application icon

### ISSUE THREAD

**Issue by malankriel** [AutoActuary/app-builder#24]
When no icon argument is specified:
![image](https://user-images.githubusercontent.com/51300197/156351944-ef8d9374-74af-49d8-93a7-a2848e29ed56.png)

I think we should default to the application icon:
![image](https://user-images.githubusercontent.com/51300197/156353115-4abe55ad-bf45-4f6a-a4d9-5e18c80f6331.png)

But it uses the system default for the filetype extension:
![image](https://user-images.githubusercontent.com/51300197/156353057-8729cec0-04f9-451b-ba84-7a43972dc1cc.png)

**Comment by heetbeet** [comment]
Good idea, only one slight issue, the .ico is then required to be shipped with the program (i.e. this will have to be implicit, since the whole point is to not get stuck with an unknown icon). Currently the assumption is that the icons referenced in the list needs to be on the user's computer (either known Windows icons or icons explicitly shipped with our program). I think one way to overcome this is to store the icon in like a hash directory if not shipped with the app.

**Comment by rudolfbyker** [comment]
/blikbrein Check if this issue is done. If not, plan the required changes..

**Comment by heetbeet** [comment]
Blikbrein checked #24 against the current `1.x` default branch. Verdict: **not fully done**; keep the issue open.

What is already implemented:

- [`build.py`](https://github.com/AutoActuary/app-builder/blob/1.x/app_builder/build.py#L98) defaults each omitted shortcut icon to `installer.icon` in the generated manifest.
- [`installer_bundle.py`](https://github.com/AutoActuary/app-builder/blob/1.x/app_builder/installer_bundle.py#L740) resolves that manifest path under the installed application and [`New-AppBuilderShortcut`](https://github.com/AutoActuary/app-builder/blob/1.x/app_builder/installer_bundle.py#L722) sets `IconLocation` when the file exists.
- The current `1.x` head (`9cd5b98`) has green Test, Black, and mypy workflow runs.

Remaining gap:

- Payload contents still come only from `installer.paths.include`; `installer.icon` is not added implicitly.
- The existing regression test, [`test_build_release_uses_installer_icon_for_shortcuts_and_exe`](https://github.com/AutoActuary/app-builder/blob/1.x/test/test_end_to_end.py#L130), explicitly lists `app.ico` in `paths.include`. If a project omits that workaround, the icon is absent after installation, the `Test-Path` guard skips `IconLocation`, and Windows falls back to the target file type's icon. This is the exact implicit-shipping concern raised earlier in this issue.
- A remapped icon has a related defect: the manifest records the source path rather than its installed destination.

Required changes:

1. During `build_release`, ensure `installer.icon` is in the payload whenever a Start Menu entry relies on it. Reuse its existing remapped destination when already included; otherwise stage it at a reserved stable/hash-derived payload path so exclusions and naming collisions cannot silently remove it.
2. Write that installed payload destination into defaulted shortcut manifest entries. Preserve an explicit per-shortcut `icon` override.
3. Update the regression test to remove `app.ico` from `paths.include`, then assert that the payload contains the implicitly staged icon and the manifest points to that installed path. Add coverage for an already-included/remapped icon so it is neither duplicated nor referenced by its stale source path.
4. Extend a Windows installer test to open the generated `.lnk` and verify `IconLocation`, then document that `installer.icon` is packaged automatically for defaulted Start Menu shortcuts.

No repository changes were made; this was a status audit and implementation plan.
